src/proto/media/file/dir.rs

```diff
@@ -58,16 +58,17 @@ impl<'a> Directory<'a> {
     /// * `uefi::Status::DEVICE_ERROR`       The device reported an error, the file was deleted,
     ///                                      or the end of the file was reached before the `read()`.
     /// * `uefi::Status::VOLUME_CORRUPTED`   The filesystem structures are corrupted
-    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry.
-    pub fn read_entry(
+    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry,
+    ///                                      the required buffer size is provided into the error.
+    pub fn read_entry<'buf>(
         &mut self,
-        buffer: &mut [u8],
-    ) -> result::Result<Option<&'a mut FileInfo>, (Status, usize)> {
+        buffer: &'buf mut [u8],
+    ) -> Result<Option<&'buf mut FileInfo>, Option<usize>> {
```

Comment:
I made a mistake here and this API was actually unsound. Gotta be careful with lifetime inference in unsafe code... Same problem for get_info.